federation/pkg/federatedtypes/adapter.go

```diff
@@ -54,6 +54,15 @@ type FederatedTypeAdapter interface {
 	ClusterWatch(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (watch.Interface, error)
 
 	NewTestObject(namespace string) pkgruntime.Object
+
+	// ImplementsReconcileHook is used to explicitly state, if the adapter implements ReconcilePlugin interface also.
+	ImplementsReconcilePlugin() bool
```

Comment:
Its actually the code in another PR https://github.com/kubernetes/kubernetes/pull/45497.